tests/lib/k8s/test_builders.py

```diff
@@ -42,6 +42,22 @@ def check_namespace(parameters):
             "namespace": check_namespace,
         }
 
+    @patch("nestor_api.lib.k8s.builders.Compiler", autospec=True)
+    @patch("nestor_api.lib.k8s.builders.io", autospec=True)
+    def test_load_templates(self, io_mock, template_compiler_mock):
+        io_mock.read.side_effect = lambda x: f"file: {x}"
+        compile_mock = template_compiler_mock.return_value.compile
+
+        templates = k8s_builders.load_templates("/path", ["template_a", "template_b"])
+
+        self.assertEqual(list(templates.keys()), ["template_a", "template_b"])
+        io_mock.read.assert_has_calls(
+            [call("/path/template_a.yaml"), call("/path/template_b.yaml")]
+        )
+        compile_mock.assert_has_calls(
+            [call("file: /path/template_a.yaml"), call("file: /path/template_b.yaml")]
+        )
+
```

Comment:
We could also try to add one more test where we use a template from fixtures and we try to use it after compilation to ensure it actually fits our needs and lock this behavior